### PR TITLE
adding verifyMode param to safeRequest

### PR DESCRIPTION
**Pull request by miki725**
docker in some cases needs to make a call via TLS however not validate the certificate

### CODE REVIEW

File: nimutils/net.nim [hunk 1]
```diff
@@ -177,7 +177,7 @@ proc safeRequest*(client: HttpClient,
 
 # https://github.com/nim-lang/Nim/blob/a45f43da3407dbbf8ecd15ce8ecb361af677add7/lib/pure/httpclient.nim#L380-L386
 # similar to stdlib but defaults to bundled CAs
-proc getSSLContext(caFile: string = ""): SslContext =
+proc getSSLContext(caFile: string = "", verifyMode = CVerifyPeer): SslContext =
```
Comment (ee7) [inline nimutils/net.nim:180]: This `verifyMode` parameter is unused. Let's either change the `verifyMode = CVerifyPeer` uses in this proc to `verifyMode = verifyMode`, or remove this parameter.

**Review by ee7 — DISMISSED** [review]
The new `getSSLContext` parameter is unused, so the `CVerifyPeer` mode is always used even if the caller tries to use a different mode. This PR writes:

```nim
context = getSSLContext(caFile = pinnedCert, verifyMode = verifyMode)
```

but then it only uses the `CVerifyPeer` mode because that's hardcoded inside `getSSLContext`.

File: nimutils/net.nim (hunk 1 above)
Comment (miki725) [reply nimutils/net.nim:180]: 🤦 ooops. pushed fix

Comment (ee7) [reply nimutils/net.nim:180]: Confirming resolved in https://github.com/crashappsec/nimutils/compare/4150fd22bd77de37d6cbc8233c65e3cfb61eb849..43049365fce3ce7b5c1556747c023d56da457385

**Review by ee7 — COMMENTED** [review]
This PR added `verifyMode` as a new parameter for `createHttpClient`, but did not alter `newAwsClient` proc, which uses `createHttpClient`:
 https://github.com/crashappsec/nimutils/blob/43049365fce3ce7b5c1556747c023d56da457385/nimutils/awsclient.nim#L95-L105

Should we add a `verifyMode` parameter to `newAwsClient`?

**Comment by miki725** [comment]
aws client always needs to verify its peer. no need to customize it

**Review by ee7 — APPROVED** [review]

**Pull request merged**